contract/src/Custody.sol

```diff
@@ -373,8 +422,15 @@ contract Custody is IChannel, IDeposit {
         uint256 correctedAmount = channelBalance > amount ? amount : channelBalance;
         meta.tokenBalances[token] -= correctedAmount;
 
-        Account storage accountInfo = _accounts[account];
-        accountInfo.tokens[token].locked -= correctedAmount;
-        accountInfo.tokens[token].available += correctedAmount;
+        Ledger storage accountInfo = _accounts[account];
+
+        // Check locked amount before subtracting to prevent underflow
```

Comment:
I think this logic is redundant as this situation should not be possible. Still, if this "safeguard" is a requirement, let's move the line `accountInfo.tokens[token].available += amountToUnlock` out from the `if` statement, as channel balance (`meta.tokenBalances[token] -= correctedAmount`) is decreased regardless, so available balance should increase regardless too.

Reply:
Ok, we will have to test well those method on edge cases

Reply:
90d201e7b0cd953cd9cc6c659c53176262a2e090